File: choijinhap/11265.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main_11265 {
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringTokenizer st = new StringTokenizer(br.readLine());
		int N = Integer.parseInt(st.nextToken());
		int M = Integer.parseInt(st.nextToken());
		int[][] map = new int[N + 1][N + 1];
		for (int i = 1; i <= N; i++) {
			st = new StringTokenizer(br.readLine());
			for (int j = 1; j <= N; j++) {
				map[i][j] = Integer.parseInt(st.nextToken());
			}
		}
		
		//플로이드와샬알고리즘 으로 모든 파티장에서 모든 파티장으로의 최단거리 계산
		for (int k = 1; k <= N; k++) {
			for (int i = 1; i <= N; i++) {
				for (int j = 1; j <= N; j++) {
					if (map[i][k] + map[k][j] < map[i][j])
						map[i][j] = map[i][k] + map[k][j];
				}
			}
		}
		
		for(int i=0;i<M;i++) {
			st=new StringTokenizer(br.readLine());
			int A=Integer.parseInt(st.nextToken());
			int B=Integer.parseInt(st.nextToken());
			int C=Integer.parseInt(st.nextToken());
			if(map[A][B]<=C) System.out.println("Enjoy other party");
			else System.out.println("Stay here");
		}
	}
}

File: Yeom/11265.py
from collections import deque
import sys

N,M=map(int,sys.stdin.readline().split());
arr=list();
for _ in range(N):
  arr.append(list(map(int,sys.stdin.readline().split())));

# 경 출 도
for k in range(N):
  for s in range(N):
    for e in range(N):
      if arr[s][e]>arr[s][k]+arr[k][e]:
        arr[s][e]=arr[s][k]+arr[k][e];

for _ in range(M):
  A,B,C=map(int,sys.stdin.readline().split());
  if arr[A-1][B-1]<=C:
    print("Enjoy other party");
  else:
    print("Stay here");